src/main/flight/imu.c

```diff
@@ -342,3 +342,14 @@ int16_t calculateThrottleAngleCorrection(uint8_t throttle_correction_value)
         angle = 900;
     return lrintf(throttle_correction_value * sinf(angle / (900.0f * M_PIf / 2.0f)));
 }
+
+int16_t calculateThrottleAngleCorrectionAxis(uint8_t throttle_hovering_percent, uint8_t axis){
+
+    if (ABS(anglerad[axis]) > M_PIf/2)
+        return 0; //if upside down don't increase correction
+
+    uint16_t liftOffTrust = throttle_hovering_percent * 10; //to bring from 0-100 range to 0-1000 range (thrust input is in the 1000-2000 range)
+    int16_t throttleCompensation = lrintf( liftOffTrust * cosf( ABS(anglerad[axis]) ) - liftOffTrust );
```

Comment:
Again, this doesn't look correct. You will end up with the negative compensation. Have you flown this? Have you even tested it on the controller on the bench? Please, please, please, don't waste your time on the "compensation". The only way to actually compensate is to divide the current throttle by cos(angle), where angle is between "down for the quad" and "down to earth", with some way of smoothing it out at the high angles. If you're not doing exactly that, I wouldn't like this code to go into the official repo, because it doesn't make the current situation (which, at least, doesn't make you to compensate more manually) any better. It may be doing what you want it to do, but it's not what you state in the description and not what most people need to happen to their throttle inflight.